clang/test/Driver/sycl-offload-amdgcn.cpp

```diff
@@ -0,0 +1,36 @@
+/// Tests specific to `-fsycl-targets=amdgcn-amd-amdhsa-sycldevice`
+// REQUIRES: clang-driver
+
+// UNSUPPORTED: system-windows
+
+/// Check action graph.
+// RUN: %clangxx -### -std=c++11 -target x86_64-unknown-linux-gnu -fsycl \
+// RUN: -fsycl-targets=amdgcn-amd-amdhsa-sycldevice -mcpu=gfx906 \
+// RUN: -fsycl-libspirv-path=%S/Inputs/SYCL/libspirv.bc %s 2>&1 \
```

Comment:
The remaining failure may well be due to the placement of the `-###` argument (it's effect is different based on the position on the CLI):
```suggestion
// RUN: %clangxx -std=c++11 -target x86_64-unknown-linux-gnu -fsycl \
// RUN: -fsycl-targets=amdgcn-amd-amdhsa-sycldevice -mcpu=gfx906 \
// RUN: -fsycl-libspirv-path=%S/Inputs/SYCL/libspirv.bc %s -### 2>&1 \
```
Please check if this works locally; LGTM otherwise.

Reply:
I'm not clear why change the placement of the `-###`, I refer to the code from `sycl-offload-nvptx.cpp`.

Reply:
The reason I've considered `-###` placement is that, if preceding other arguments, it only dumps option parsing stage results (https://clang.llvm.org/docs/DriverInternals.html#driver-stages).
Judging by the failure logs:
```         
clang-13: error: cannot find ROCm device library. Provide its path via --rocm-path or --rocm-device-lib-path, or pass -nogpulib to build without ROCm device library. 
check:11     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
clang-13: error: cannot find ROCm device library. Provide its path via --rocm-path or --rocm-device-lib-path, or pass -nogpulib to build without ROCm device library.
```
Maybe configuring `-nogpulib` would also help duck the failure.

Reply:
How can I configuring `-nogpulib` option?  And how can I obtain the failure logs above?

Reply:
Please see http://ci.llvm.intel.com:8010/#builders/2/builds/14878 for the latest logs - this is accessible from the list of checks.

Reply:
The `-nolibgpu` addition, as shown in the current suggestions, helped me uncover actual failure causes, but after these are fixed, `-nolibgpu` is no longer needed. In other comments, In another comment, I'll add a patch that makes the test pass locally for me.
```suggestion
// RUN: %clangxx -### -std=c++11 -target x86_64-unknown-linux-gnu -fsycl \
// RUN: -fsycl-targets=amdgcn-amd-amdhsa-sycldevice -mcpu=gfx906 \
// RUN: -fsycl-libspirv-path=%S/Inputs/SYCL/libspirv.bc %s 2>&1 \
```